liblava/resource/mesh.hpp

```diff
@@ -74,6 +74,125 @@ struct mesh_data {
     }
 };
 
+/**
+ * @brief Temporary templated mesh data
+ *
+ * @tparam T Input vertex struct
+ */
+template<typename T = vertex>
+struct generic_mesh_data {
+    /// List of vertices
+    std::vector<T> vertices;
+
+    /// List of indices.
+    index_list indices;
+
+    /**
+     * @brief Move mesh data by offset
+     *
+     * @tparam PosVecType Type of coordinate vector
+     * @tparam PosType    Type of coordinate element
+     *
+     * @param offset      Position offset
+     * @param factor      Byte offset of coordinate vector within T
+     */
+    template<typename PosVecType = lava::v3, typename PosType = float>
+    void move(std::array<PosType, 3> offset, size_t struct_pos_offset = 0) {
+        for (T& vertex : vertices) {
+            for (size_t i = 0; i < 3; i++) {
+                (*((std::array<PosType, 3>*) (&vertex + struct_pos_offset)))[i] += offset[i];
+            }
+        }
+    }
+
+    /**
+     * @brief Scale mesh data by factor
+     *
+     * @tparam PosType  Type of coordinate element
+     *
+     * @param factor    Position scaling factor
+     * @param factor    Byte offset of coordinate vector within T
+     */
+    template<typename PosType = float>
+    void scale(PosType factor, size_t struct_pos_offset = 0) {
+        for (T& vertex : vertices) {
+            for (size_t i = 0; i < 3; i++) {
+                (*((std::array<PosType, 3>*) (&vertex + struct_pos_offset)))[i] *= factor;
+            }
+        }
+    }
+
+private:
+    static constexpr bool const is_default = std::is_same_v<T, vertex>;
```

Comment:
Originally it was used in a `constexpr` expression to statically error on using `scale` or `move` in non lava::vertex meshes, but then I got the idea to implement them with pointer arithmetic. I left it in because I imagined it might be useful at some point. I can remove it, since I don't have any specific ideas what it might be useful for.

Reply:
All right. I didn't know if you wanted to use it later 👍

Reply:
I've now removed it.